### PR TITLE
[INFINITY-2184] Style fixes in some universe json files

### PR DESCRIPTION
**Pull request by mbdcos**

### CODE REVIEW

File: frameworks/helloworld/universe/config.json [hunk 1]
```diff
@@ -49,105 +49,105 @@
...
+            "default": ""
           },
           "secret2": {
             "description": "secret2",
             "type": "string",
             "default": ""
           }
         },
-        "required":[
+        "required": [
```
Comment (nickbp) [inline frameworks/helloworld/universe/config.json]: trailing space

File: frameworks/hdfs/universe/config.json [hunk 2]
```diff
@@ -141,35 +141,35 @@
           "default": true
         }
       },
-      "required":[
+      "required": [
```
Comment (nickbp) [inline frameworks/hdfs/universe/config.json]: trailing space

File: frameworks/kafka/universe/config.json [hunk 3]
```diff
@@ -1,665 +1,665 @@
...
             "type": "string",
-            "description": "The user that the service will run as.",
-            "default": "nobody"
+            "default": "V0"
           },
-          "principal": {
-            "description": "The principal for the Kafka service instance.",
-            "type": "string",
-            "default": ""
+          "user":  {
```
Comment (nickbp) [inline frameworks/kafka/universe/config.json]: double space

File: frameworks/kafka/universe/config.json [hunk 4]
```diff
@@ -1,665 +1,665 @@
...
+            "type":  "string",
+            "description":  "The user that the service will run as.",
+            "default":  "nobody"
           },
-          "secret_name": {
-            "description": "Name of the Secret Store credentials to use for DC/OS service authentication. This should be left empty unless service authentication is needed.",
-            "type": "string",
-            "default": ""
+          "principal":  {
```
Comment (nickbp) [inline frameworks/kafka/universe/config.json]: double space

File: frameworks/kafka/universe/config.json [hunk 5]
```diff
@@ -1,665 +1,665 @@
...
+            "description":  "The principal for the Kafka service instance.",
+            "type":  "string",
+            "default":  ""
           },
-          "placement_constraint":{
-            "description":"Marathon-style placement constraint for Broker nodes. Example: rack_id:LIKE:rack-foo-.*,rack_id:MAX_PER:2",
-            "type":"string",
-            "default":"hostname:MAX_PER:1"
+          "secret_name":  {
```
Comment (nickbp) [inline frameworks/kafka/universe/config.json]: double space

File: frameworks/kafka/universe/config.json [hunk 6]
```diff
@@ -1,665 +1,665 @@
...
+            "description":  "Name of the Secret Store credentials to use for DC/OS service authentication. This should be left empty unless service authentication is needed.",
+            "type":  "string",
+            "default":  ""
```
Comment (nickbp) [inline frameworks/kafka/universe/config.json]: double space

File: frameworks/kafka/universe/config.json [hunk 7]
```diff
@@ -1,665 +1,665 @@
...
+            "description":  "The principal for the Kafka service instance.",
+            "type":  "string",
+            "default":  ""
           },
-          "placement_constraint":{
-            "description":"Marathon-style placement constraint for Broker nodes. Example: rack_id:LIKE:rack-foo-.*,rack_id:MAX_PER:2",
-            "type":"string",
-            "default":"hostname:MAX_PER:1"
+          "secret_name":  {
+            "description":  "Name of the Secret Store credentials to use for DC/OS service authentication. This should be left empty unless service authentication is needed.",
```
Comment (nickbp) [inline frameworks/kafka/universe/config.json]: double space

File: frameworks/kafka/universe/config.json [hunk 8]
```diff
@@ -1,665 +1,665 @@
...
+            "type":  "string",
```
Comment (nickbp) [inline frameworks/kafka/universe/config.json]: double space

File: frameworks/kafka/universe/config.json [hunk 9]
```diff
@@ -1,665 +1,665 @@
...
           },
-          "log_level": {
-            "description": "The log level for the DC/OS service.",
+          "deploy_strategy": {
+            "description": "Deployment phase strategy. See documentation. [serial, serial-canary, parallel-canary, parallel]",
             "type": "string",
-            "enum": ["OFF", "FATAL", "ERROR", "WARN", "INFO", "DEBUG", "TRACE", "ALL"],
-            "default": "INFO"
+            "default": "serial",
+            "enum":  ["serial", "serial-canary", "parallel-canary", "parallel"]
```
Comment (nickbp) [inline frameworks/kafka/universe/config.json]: double space

File: frameworks/kafka/universe/config.json [hunk 10]
```diff
@@ -1,665 +1,665 @@
...
+          },
+          "log_level":  {
```
Comment (nickbp) [inline frameworks/kafka/universe/config.json]: lots of double spaces

File: frameworks/kafka/universe/config.json [hunk 11]
```diff
@@ -1,665 +1,665 @@
...
+              "size":  {
+                "type": "integer",
+                "description": "The amount of JVM heap, in MB, allocated to the Kafka broker process.",
+                "default": 512
               }
             },
-            "additionalProperties": false,
-            "required": [
+            "additionalProperties":  false,
+            "required":  [
```
Comment (nickbp) [inline frameworks/kafka/universe/config.json]: double space

File: frameworks/kafka/universe/config.json [hunk 12]
```diff
@@ -1,665 +1,665 @@
...
+            "properties": {
+              "size":  {
+                "type": "integer",
+                "description": "The amount of JVM heap, in MB, allocated to the Kafka broker process.",
+                "default": 512
               }
             },
-            "additionalProperties": false,
-            "required": [
+            "additionalProperties":  false,
```
Comment (nickbp) [inline frameworks/kafka/universe/config.json]: double space

File: frameworks/kafka/universe/config.json [hunk 13]
```diff
@@ -1,665 +1,665 @@
...
-          "disk":{
-            "description":"Broker disk requirements (only respected with persistent volumes)",
-            "type":"integer",
-            "default":5000
+          "disk": {
+            "description": "Broker disk requirements (only respected with persistent volumes)",
+            "type": "integer",
+            "default": 5000
+          },
+          "disk_type":  {
```
Comment (nickbp) [inline frameworks/kafka/universe/config.json]: double spaces throughout, here and below

**Comment by elezar** [comment]
We could run all the json files through a formatter, either `jq` or `python3 -m json.tool`. We could even add this as a pre-commit hook.

Any objections, or are there specifics to our json files?


1. Create a file `format_json.sh` with the following contents:
```bash
#!/bin/bash
input_filename=$1
python3 -m json.tool "$input_filename" "$input_filename.tmp"
mv "$input_filename.tmp" "$input_filename"
git add "$input_filename"
```
2. Set the permissions to `+x`: `chmod +x format_json.sh`
3. And then run:
```bash
find . -name "config.json" -exec ./check_json.sh {} \;
```

This will add all the formatted `config.json` files to the git index. These can be committed and pushed.

Thoughts?

**Comment by nickbp** [comment]
I think that'd make sense.

Don't users need to manually enable the commit hook in their local git env?

Could have a lint call in the build telling the user what to do.

File: frameworks/kafka/universe/config.json [hunk 14]
```diff
@@ -487,17 +487,17 @@
...
-            "title": "inter.broker.protocol.version",
-            "description": "Specify which version of the inter-broker protocol will be used, which must align with log.message.format.version. This is typically bumped after all brokers were upgraded to a new version. Example of some valid values are: 0.8.0, 0.8.1, 0.8.1.1, 0.8.2, 0.8.2.0, 0.8.2.1, 0.9.0.0, 0.9.0.1, 0.10.0.0. Check ApiVersion for the full list.",
-            "default": "0.10.0.0"
-          },
-          "log_message_format_version": {
-            "type": "string",
-            "title": "log.message.format.version",
-            "description": "Specify which version of the log message format will be used, which must align with inter.broker.protocol.version. This is a new setting as of 0.10.0.0, and should be left at 0.9.0 until clients are updated to 0.10.0.x. Clients on earlier versions may see a performance penalty if this is increased before they've upgraded. See the latest Kafka documentation for details.",
-            "default": "0.10.0"
+          "inter_broker_protocol_version":{
```
Comment (elezar) [inline frameworks/kafka/universe/config.json]: @mbdcos the formatting changes that were applied in this PR seem to have been removed.

Comment (mbdcos) [inline frameworks/kafka/universe/config.json]: ¯\_(ツ)_/¯

**Comment by mbdcos** [comment]
Since we know HDFS CI is broken, I am thinking to merge this PR. It has been waiting for a long time, and it is a config changes, so always getting merge conflicts when other PRs are in.  Or alternatively, we can close this PR and forget about this change. I do not think it worths the effort to keep fixing merge conflicts and babysitting CI.

**Comment by benclarkwood** [comment]
@mbdcos Let's wait for the HDFS disabling to occur, then we can get this in.

**Comment by mbdcos** [comment]
All green 👍

**Pull request merged**